app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -44,6 +44,7 @@ class Bolt_Boltpay_Model_Payment extends Mage_Payment_Model_Method_Abstract
     const TRANSACTION_REJECTED_REVERSIBLE = 'rejected_reversible';
     const TRANSACTION_REJECTED_IRREVERSIBLE = 'rejected_irreversible';
     const TRANSACTION_NO_NEW_STATE = 'no_new_state';
+    const TRANSACTION_REFUND = 'credit';
```

Comment:
Yes, there is no transaction type of refund, but since we have a mapping 

```
public static $_hookTypeToStatusTranslator = array(
        self::HOOK_TYPE_AUTH => self::TRANSACTION_AUTHORIZED,
        self::HOOK_TYPE_CAPTURE => self::TRANSACTION_COMPLETED,
        self::HOOK_TYPE_PAYMENT => self::TRANSACTION_COMPLETED,
        self::HOOK_TYPE_PENDING => self::TRANSACTION_PENDING,
        self::HOOK_TYPE_REJECTED_REVERSIBLE => self::TRANSACTION_REJECTED_REVERSIBLE,
        self::HOOK_TYPE_REJECTED_IRREVERSIBLE => self::TRANSACTION_REJECTED_IRREVERSIBLE,
        self::HOOK_TYPE_VOID => self::TRANSACTION_CANCELLED,        
        self::HOOK_TYPE_REFUND => self::TRANSACTION_REFUND
    );
```
then I create *self::TRANSACTION_REFUND* to adapt this, and I think it is rather than a placeholder

Reply:
and maybe on bolt server side, `the correct transaction status after a refund is still const TRANSACTION_COMPLETED = 'completed' ` , but on merchant side, I think we would better separate them,  if not, it is hard to distinguish capture from refund.

Reply:
ok.  it does make the code read more clearly.

Reply:
sounds good to me!